src/System/God/GodDoodle.ZC

```diff
@@ -0,0 +1,192 @@
+#help_index "God;Graphics/Sprite;Sprites"
+
+U0 GodDoodleDraw(CTask *task,CDC *dc)
+{
+  GrBlot(dc,0,0,god.doodle_dc);
+  if (Blink) {
+    if (god.doodle_done) {
+      dc->color=RED;
+      GrPrint(dc,(task->pix_width-FONT_WIDTH*29)>>1,
+	    (task->pix_height-3*FONT_HEIGHT)>>1,
+	    "Press <ESC> to insert sprite.");
+      GrPrint(dc,(task->pix_width-FONT_WIDTH*39)>>1,
+	    (task->pix_height-3*FONT_HEIGHT)>>1+2*FONT_HEIGHT,
+	    "Press <SHIFT-ESC> to throw-away sprite.");
+    } else {
+      dc->color=GREEN;
+      GrPrint(dc,(task->pix_width-FONT_WIDTH*25)>>1,
+	    (task->pix_height-FONT_HEIGHT)>>1,
+	    "Press <SPACE> repeatedly.");
+    }
+  }
+}
+
+U0 GodDoodleSmooth(I64 num)
+{
+  CDC *dc=DCExt(god.doodle_dc,0,0,
+	god.doodle_dc->width-1,god.doodle_dc->height-1);
+  I64 i,x,y,x1,y1,c,histogram[16],best,best_cnt,c_old=god.doodle_dc->color;
+  for (y=0;y<god.doodle_dc->height;y++)
+    for (x=0;x<god.doodle_dc->width;x++) {
+      MemSet(histogram,0,sizeof(histogram));
+      for (y1=y-num;y1<=y+num;y1++)
+	for (x1=x-num;x1<=x+num;x1++) {
+	  c=GrPeek(dc,x1,y1);
+	  if (0<=c<=15)
+	    histogram[c]++;
+	}
+      best=BLACK;
+      best_cnt=-1;
+      for (i=0;i<16;i++)
+	if (histogram[i]>best_cnt) {
+	  best=i;
+	  best_cnt=histogram[i];
+	}
+      god.doodle_dc->color=best;
+      GrPlot(god.doodle_dc,x,y);
+    }
+  god.doodle_dc->color=c_old;
+  DCDel(dc);
+}
+
+U0 GodDoodleBitsIns(I64 num_bits,I64 n)
+{//Insert bits into God doodle bit fifo.
+  I64 i;
+  for (i=0;i<num_bits;i++) {
+    FifoU8Insert(god.doodle_fifo,n&1);
+    n>>=1;
+  }
+}
+
+U0 GodDoodleHexIns(U8 *st)
+{//Insert hex record into God doodle bit fifo.
+  U8 buf[2];
+  if (st) {
+    buf[1]=0;
+    while (*buf=*st++)
+      if (Bt(char_bmp_hex_numeric,*buf))
+	GodDoodleBitsIns(4,rev_bits_table[Str2I64(buf,16)]>>4);
```

Comment:
The indent is wrong
See:
https://github.com/cia-foundation/TempleOS/blob/c26482bb6ad3f80106d28504ec5db3c6a360732c/Adam/God/GodDoodle.HC#L67-L68